clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -2548,6 +2548,23 @@ public void testRaiseErrorWhenNoPartitionsPendingOnDrain() throws InterruptedExc
         // We shouldn't drain batches which haven't been added to the transaction yet.
         assertTrue(drainedBatches.containsKey(node1.id()));
         assertTrue(drainedBatches.get(node1.id()).isEmpty());
+
+        // Let's now add the partition, flush and try to drain again.
+        transactionManager.maybeAddPartition(tp0);
+        accumulator.beginFlush();
+
+        drainedBatches = accumulator.drain(metadataMock, nodes, Integer.MAX_VALUE,
```

Comment:
@artemlivshits There seems to be an error in this line. The variable `metadataMock` is not defined and that causes the test to break. I think the variable you were looking for is `metadataCache`. Could you please check the same?
Thanks.

Reply:
@sjhajharia I also experienced this issue. Variable `metadataMock` was probably removed after the CI build was run and the PR was merged. The CI build for this PR does not have a compilation error.
The PR that removed the variable is https://github.com/apache/kafka/commit/ff90f78c700c582f9800013faad827c36b45ceb7#diff-36184945d0619b12edaf88d58f6f7668d543cbc534948d381017eb978a033a59

I will try to fix this.

Reply:
This PR should fix the compilation error: https://github.com/apache/kafka/pull/15405

Reply:
thanks folks